x/tokenregistry/keeper/keeper.go

```diff
@@ -165,3 +165,46 @@ func (k keeper) SetFirstLockDoublePeg(ctx sdk.Context, denom string, networkDesc
 		instrumentation.PeggyCheckpoint(ctx.Logger(), instrumentation.SetFirstLockDoublePeg, "networkDescriptor", networkDescriptor, "registry", registry)
 	}
 }
+
+func (k keeper) AddMultipleTokens(ctx sdk.Context, entries []*types.RegistryEntry) {
+	wl := k.GetRegistry(ctx)
+
+	for _, entry := range entries {
+		found := false
+		for i := range wl.Entries {
+			// update if nil or the same denom
+			if wl.Entries[i] != nil && strings.EqualFold(wl.Entries[i].Denom, entry.Denom) {
```

Comment:
Is it compulsory to have to write all this logic manually in Go? It is a common task to compare a parameter of an element in a list of things against a key in order to avoid inserting a duplicate. Also, if duplicates are unwanted, why not make `wl.Entries` into a map keyed by `Denom`? Does Go have a collections library for this sort of thing?

These lists better not be too large either, because this is `O(m * n)`.

Reply:
a map was implemented before, and solution was reverted after the last merge with develop branch. will recover the solution later.